### DISCUSSION TITLE
Difference between `Head` in `next/document` and `next/head`

### DISCUSSION THREAD

**Discussion by someguynamedmatt** [vercel/next.js#12290] [Help]
Hello Next community,

I was messing around and noticed I couldn't put a `script` tag as a child to the `next/head` component (`<Head />`), but I could place one in `next/document`'s. I dug through the source slightly and noticed the [`next/document`](https://github.com/zeit/next.js/blob/f0809316eec7eea09781a71cf514c0bacee4d7d2/packages/next/pages/_document.tsx#L348) parses the script tags, whereas [`next/head`](https://github.com/zeit/next.js/blob/f0809316eec7eea09781a71cf514c0bacee4d7d2/packages/next/next-server/lib/head.tsx#L57) only seems to parse `meta`, `title`, and `base`. 

Can anyone explain why these appear to behave differently? The discussions I've come across online seem to suggest that `next/head` is preferred as a replacement to something like `react-helmet`, so why the disparity between the two exports? Maybe I'm misunderstanding the use case for these two components and, if so, clarification would be awesome!

**Comment by majelbstoat** [comment]
I'm also confused by this. I believe _document - and thus document/head - is rendered statically on the server side, whereas next/head used in _app, and pages, loads serverside and clientside.

But I'm confused by which kinds of tags should go where. If I put a viewport tag in document/head, I end up with a duplicate. If I put it in an _app.tsx next/head, I don't - prolly because it's dynamically rendered and replaces the default. But it's not the kind of thing I want re-rendered on every page load (same with rel preloads, twitter:site_name meta stuff etc). Clearer documentation on best practices would be great.

**Comment by TyMick** [comment]
I just started wondering the same thing when trying to fix a [multiple viewport meta tags](https://github.com/zeit/next.js/issues/13230) issue. I'm thinking that maybe adding children to `next/document`'s `Head` is never a good idea.

@someguynamedmatt, adding a `script` child to `next/head`'s `Head` seems to be working for me. Just made a new `create-next-app`, and when I add the following `pages/_app.js` the script shows up on the page.

```jsx
import Head from "next/head";

function MyApp({ Component, pageProps }) {
  return (
    <>
      <Head>
        <script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossOrigin="anonymous" />
      </Head>
      <Component {...pageProps} />
    </>
  );
}

export default MyApp;
```

Also works if I add the script to the `Head` component in `index.js` instead. I don't think [that `unique()` function in `head.tsx`](https://github.com/zeit/next.js/blob/canary/packages/next/next-server/lib/head.tsx#L52-L112) only parses `meta`, `title`, and `base` tags—those are just the only ones it checks for duplicates.

@majelbstoat, does stuff in `_app.js` really re-render on every page load? I though the point of `_app.js` was for things to persist between page changes. This may be a lack of understanding on my part of how React works at a lower level, though.

**Reply by majelbstoat** [reply]
From the _document documentation: "Document is only rendered in the server, event handlers like onClick won't work"

From the _app documentation: "The Component prop is the active page, so whenever you navigate between routes, Component will change to the new page."

Because Component changes, MyApp would be re-rendered

**Comment by uinstinct** [comment]
Is adding the next/head better in the index.js or the _app.js (considering the application to be only a single page) ?

**Reply by TyMick** [reply]
If the application's only a single page, you may not need an `_app.js` altogether. 🤷🏼‍♂️ The only context I'm aware of where using `next/head` in `_app.js` is better is to prevent yourself from having to repeat the same code on multiple pages.

**Comment by seho0808** [comment]
I think the official document got it. 

**_"The `<Head />` component used here should only be used for any `<head>` code that is common for all pages. For all other cases, such as `<title>` tags, we recommend using next/head in your pages or components."_**
(Here, `<Head />` refers to next/document head.)

from: https://nextjs.org/docs/advanced-features/custom-document

Also the official doc(https://nextjs.org/docs/api-reference/next/head) mentions that `<script>` can be parsed by next/head.
They might have changed it over time?
